donkeycar/vehicle.py

```diff
@@ -154,20 +155,26 @@ def start(self, rate_hz=10, max_loop_count=None, verbose=False):
                 self.update_parts()
 
                 # stop drive loop if loop_count exceeds max_loopcount
-                if max_loop_count and loop_count > max_loop_count:
+                if max_loop_count and loop_count >= max_loop_count:
                     self.on = False
-
-                sleep_time = 1.0 / rate_hz - (time.time() - start_time)
-                if sleep_time > 0.0:
-                    time.sleep(sleep_time)
                 else:
-                    # print a message when could not maintain loop rate.
-                    if verbose:
-                        logger.info('WARN::Vehicle: jitter violation in vehicle loop '
-                              'with {0:4.0f}ms'.format(abs(1000 * sleep_time)))
+                    sleep_time = 1.0 / rate_hz - (time.time() - start_time)
+                    if sleep_time > 0.0:
+                        time.sleep(sleep_time)
+                    else:
+                        # print a message when could not maintain loop rate.
+                        if verbose:
+                            logger.info('WARN::Vehicle: jitter violation in vehicle loop '
+                                  'with {0:4.0f}ms'.format(abs(1000 * sleep_time)))
+
+                    if verbose and loop_count % 200 == 0:
+                        self.profiler.report()
+
+
+            loop_total_time = time.time() - loop_start_time
+            print(f"Vehicle executed {loop_count} steps in {loop_total_time} seconds.")
```

Comment:
Optimally this should be a logger statement as well.

Reply:
I updated that to a logger statement

Reply:
updated to version="4.4.dev5"